src/runtime/schema.ts

```diff
@@ -156,34 +156,49 @@ export class Schema {
   equals(otherSchema: Schema): boolean {
     // TODO(cypher1): Check equality without calling contains.
     return this === otherSchema || (this.name === otherSchema.name
-       && this.isAtleastAsSpecificAs(otherSchema)
-       && otherSchema.isAtleastAsSpecificAs(this));
+       && (this.isEquivalentOrMoreSpecific(otherSchema) === AtLeastAsSpecific.YES)
+       && (otherSchema.isEquivalentOrMoreSpecific(this) === AtLeastAsSpecific.YES));
   }
 
-  isAtleastAsSpecificAs(otherSchema: Schema): boolean {
+  isEquivalentOrMoreSpecific(otherSchema: Schema): AtLeastAsSpecific {
     const names = new Set(this.names);
     for (const name of otherSchema.names) {
       if (!names.has(name)) {
-        return false;
+        return AtLeastAsSpecific.NO;
       }
     }
     // tslint:disable-next-line: no-any
     const fields: Dictionary<any> = {};
     for (const [name, type] of Object.entries(this.fields)) {
       fields[name] = type;
     }
+    let best = AtLeastAsSpecific.YES;
     for (const [name, type] of Object.entries(otherSchema.fields)) {
       if (fields[name] == undefined) {
-        return false;
+        return AtLeastAsSpecific.NO;
       }
       if (!Schema.typesEqual(fields[name], type)) {
-        return false;
+        return AtLeastAsSpecific.NO;
       }
-      if (Refinement.isAtleastAsSpecificAs(fields[name].refinement, type.refinement) === AtleastAsSpecific.NO) {
-        return false;
+      const fieldRes = Refinement.isAtLeastAsSpecificAs(fields[name].refinement, type.refinement);
+      if (fieldRes === AtLeastAsSpecific.NO) {
+        return AtLeastAsSpecific.NO;
+      } else if (fieldRes === AtLeastAsSpecific.UNKNOWN) {
+        best = AtLeastAsSpecific.UNKNOWN;
       }
     }
-    return Refinement.isAtleastAsSpecificAs(this.refinement, otherSchema.refinement) !== AtleastAsSpecific.NO;
+    const res = Refinement.isAtLeastAsSpecificAs(this.refinement, otherSchema.refinement);
+    if (res === AtLeastAsSpecific.NO) {
+      return AtLeastAsSpecific.NO;
+    } else if (res === AtLeastAsSpecific.UNKNOWN) {
+      best = AtLeastAsSpecific.UNKNOWN;
+    }
+    return best;
+  }
+
+  isAtleastAsSpecificAs(otherSchema: Schema): boolean {
```

Comment:
Did you want to correct this one too? `isAtLeast...`

Reply:
Thought my language server rename would have got them all. Resorted to `git grep` and found a bunch, thanks.